stl/inc/chrono

```diff
@@ -5171,6 +5177,165 @@ namespace chrono {
+        _CharT _Fill[4] = {' ', _CharT{0}, _CharT{0}, _CharT{0}};
+        // recursive definition in grammar, so could have anu number of these with literal chars
+        vector<_Chrono_specs<_CharT>> _Chrono_specs_list;
+    };
+
+    // Model of _Chrono_parse_spec_callbacks that fills a _Chrono_format_specs with the parsed data
+    template <class _CharT>
+    class _Chrono_specs_setter {
+    public:
+        explicit constexpr _Chrono_specs_setter(_Chrono_format_specs<_CharT>& _Specs_) : _Specs(_Specs_) {}
+
+        // same as _Specs_setter
+        constexpr void _On_align(_Align _Aln) {
+            _Specs._Alignment = _Aln;
+        }
+
+        // same as _Specs_setter
+        constexpr void _On_fill(basic_string_view<_CharT> _Sv) {
+            if (_Sv.size() > 4) {
+                throw format_error("Invalid fill (too long).");
+            }
+
+            _STD fill(_Specs._Fill, _Specs._Fill + 4, _CharT{});
+            _STD copy(_Sv.begin(), _Sv.end(), _Specs._Fill);
+        }
+
+        constexpr void _On_width(int _Width) {
+            _Specs._Width = _Width;
+        }
+
+        constexpr void _On_precision(int _Prec) {
+            _Specs._Precision = _Prec;
+        }
+
+        constexpr void _On_conversion_spec(_CharT _Modifier, _CharT _Type) {
+            // NOTE: same performance note from _Basic_format_specs also applies here
+            char _Char_mod  = static_cast<char>(_Modifier);
+            char _Char_type = static_cast<char>(_Type);
+            if (_Char_mod != 'E' && _Char_mod != 'O') {
+                throw format_error("Invalid modifier specification.");
+            }
+
+            if (_Type < 0 || _Type > (numeric_limits<signed char>::max)()) {
+                throw format_error("Invalid type specification.");
+            }
+
+            _Chrono_specs<_CharT> _Conv_spec{_Char_mod, _Char_type};
+            _Specs._Chrono_specs_list.push_back(_Conv_spec);
+        }
+
+        constexpr void _On_lit_char(_CharT _Lit_ch) {
+            _Chrono_specs<_CharT> _Lit_char_spec{_Lit_ch};
+            _Specs._Chrono_specs_list.push_back(_Lit_char_spec);
+        }
+
+    protected:
+        _Chrono_format_specs<_CharT>& _Specs;
+    };
+
+    template <class _CharT, _Chrono_parse_spec_callbacks<_CharT> _Callbacks_type>
+    _NODISCARD constexpr const _CharT* _Parse_conversion_specs(
+        const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+        ++_Begin; // move past %
+        if (_Begin == _End) {
+            throw format_error("Invalid format string.");
+        }
+
+        _CharT _Mod = '\0';
+        _CharT _Type;
+        _CharT _Ch = *_Begin;
+
+        if (_Ch == 'E' || _Ch == 'O') { // includes modifier
+            _Mod = _Ch;
+            ++_Begin;
+            if (_Begin == _End) {
+                throw format_error("Invalid format string - missing type after modifier.");
+            }
+        }

```

Comment:
From my understanding, `%E%d` should be an invalid format string according to the _chrono-format-specs_ grammar (the type must directly follow the modifier if it exists).  In its current state, I believe this would (1) store `E` as `_Mod`, (2) store `%` as `_Type`, and then (3) treat the `d` as a _lit-char_ in the next iteration of the while loop in `_Parse_chrono_format_spec`.  

When we work on checking the validity / applicability of different specifiers given a particular type's `struct formatter`, we will find that `%` isn't a valid `type` associated with modifier `E` and should error then.  I'm open to restructuring this, but I believe we should correctly error regardless.  I was just planning on pushing down the validity check of modifiers + types to the individual `struct formatter`s.

Reply:
Looking at `[tab:time.format.spec]`, `%` is a type, so that `"%%"` results in a `%` character. I think this ends up being valid (with an ignored `E` modifier) and will print `%d` literally - so your code is correct and there's nothing more to do.

(Unless there's Standardese somewhere saying that modifiers are an error for types that don't expect them.)